File: server/java/src/main/java/com/onlinepayments/example/service/dtos/paymentlinks/CreatePaymentLinkRequestDto.java
package com.onlinepayments.example.service.dtos.paymentlinks;

import com.onlinepayments.example.service.domain.common.enums.Currency;
import com.onlinepayments.example.service.domain.paymentlinks.Action;
import com.onlinepayments.example.service.domain.paymentlinks.ValidityPeriod;

import java.math.BigDecimal;

public record CreatePaymentLinkRequestDto(
        BigDecimal amount,
        Currency currency,
        String description,
        ValidityPeriod validFor,
        Action action
) { }

File: server/dotnet/sdk-dotnet-example/Business/DTOs/CreatePaymentLinks/CreatePaymentLinkRequestDto.cs
using Business.Domain.Common.Enums;
using Business.Domain.Enums.PaymentLinks;
using Action = Business.Domain.Enums.PaymentLinks.Action;

namespace Business.DTOs.CreatePaymentLinks;

public class CreatePaymentLinkRequestDto
{
    public decimal? Amount { get; init; }
    
    public Currency? Currency { get; init; }

    public string? Description { get; init; }

    public ValidityPeriod? ValidFor { get; init; }

    public Action? Action { get; init; }
}